How are Security Hotspot Category keys acquired?

Our team needs to use sonar-cnes-report with SonarQube 9.3; we have taken to fixing the compatibility issues ourselves. So far, the only breaks in compatibility we can find are that SonarQube has new "Security Hotspot Categories" that sonar-cnes-report cannot parse. 

We've narrowed down the means of adding support for these missing categories in `main.java.fr.cnes.sonar.report.utils.StringManager SECURITY_HOTSPOT_CATEGORIES. It seems to ensure that new categories are shown in reports, we need only provide their internal key and map it to a descriptive title to be shown in the reports.

However, we cannot locate where in the SonarQube source code that we can find the origin of these category keys. Our assumption is that sonar-cnes-report recieves data from SonarQube that contains a list of category keys that apply to that specific project.

Thus our question is this: Where are you acquiring the security hotspot category keys from? Are they provided directly from SonarQube? If not, how are you acquiring them? We're looking for the specification that will provide ALL new category keys for the latest SonarQube to bridge the compatibility gap.

@Sancretor done. I hit a couple of unit test errors: 
```
[ERROR] Failures: 
[ERROR]   AbstractFacetsProviderTest.getTimeFacets:143 expected:<44419.59709490741> but was:<44419.22209490741>
[ERROR]   DateConverterTest.validSonarQubeDateToExcelDateTest:15 expected: <44419.50796296296> but was: <44419.2162962963>
```
But these look like timing errors unrelated to my changes.

I wasn't sure how to run the linter, but I think my code would pass.

--- a/src/main/java/fr/cnes/sonar/report/utils/StringManager.java
+++ b/src/main/java/fr/cnes/sonar/report/utils/StringManager.java
@@ -209,6 +209,9 @@ public final class StringManager {
         SECURITY_HOTSPOT_CATEGORIES.put("insecure-conf", "Insecure Configuration");
         SECURITY_HOTSPOT_CATEGORIES.put("file-manipulation", "File Manipulation");
         SECURITY_HOTSPOT_CATEGORIES.put("others", "Others");
+        SECURITY_HOTSPOT_CATEGORIES.put("permission", "Permission");
+        SECURITY_HOTSPOT_CATEGORIES.put("encrypt-data", "Encryption of Sensitive Data");
+        SECURITY_HOTSPOT_CATEGORIES.put("traceability", "Traceability");
     }
 
     /**

--- a/src/test/ut/java/fr/cnes/sonar/report/exporters/data/SecurityHotspotsAdapterTest.java
+++ b/src/test/ut/java/fr/cnes/sonar/report/exporters/data/SecurityHotspotsAdapterTest.java
@@ -74,6 +74,9 @@ public class SecurityHotspotsAdapterTest {
         expected.add(Arrays.asList("Path Traversal Injection", "0", "0", "0"));
         expected.add(Arrays.asList("HTTP Response Splitting", "0", "0", "0"));
         expected.add(Arrays.asList("Others", "0", "0", "0"));
+        expected.add(Arrays.asList("Permission", "0", "0", "0"));
+        expected.add(Arrays.asList("Encryption of Sensitive Data", "0", "0", "0"));
+        expected.add(Arrays.asList("Traceability", "0", "0", "0"));
         Mockito.when(report.getToReviewSecurityHotspots()).thenReturn(listIssues);
         Assert.assertEquals(expected, SecurityHotspotsAdapter.getSecurityHotspotsByCategoryAndPriority(report));
     }

--- a/src/test/ut/java/fr/cnes/sonar/report/exporters/data/SecurityHotspotsAdapterTest.java
+++ b/src/test/ut/java/fr/cnes/sonar/report/exporters/data/SecurityHotspotsAdapterTest.java
@@ -65,7 +65,10 @@ public class SecurityHotspotsAdapterTest {
         expected.add(Arrays.asList("Log Injection", "2", "0", "0"));
         expected.add(Arrays.asList("Cross-Site Request Forgery (CSRF)", "0", "0", "0"));
         expected.add(Arrays.asList("Open Redirect", "0", "0", "0"));
+        expected.add(Arrays.asList("Permission", "0", "0", "0"));
         expected.add(Arrays.asList("SQL Injection", "0", "0", "0"));
+        expected.add(Arrays.asList("Encryption of Sensitive Data", "0", "0", "0"));
+        expected.add(Arrays.asList("Traceability", "0", "0", "0"));
         expected.add(Arrays.asList("Buffer Overflow", "0", "0", "0"));
         expected.add(Arrays.asList("File Manipulation", "0", "0", "0"));
         expected.add(Arrays.asList("Code Injection (RCE)", "0", "0", "0"));
@@ -74,9 +77,6 @@ public class SecurityHotspotsAdapterTest {
         expected.add(Arrays.asList("Path Traversal Injection", "0", "0", "0"));
         expected.add(Arrays.asList("HTTP Response Splitting", "0", "0", "0"));
         expected.add(Arrays.asList("Others", "0", "0", "0"));
-        expected.add(Arrays.asList("Permission", "0", "0", "0"));
-        expected.add(Arrays.asList("Encryption of Sensitive Data", "0", "0", "0"));
-        expected.add(Arrays.asList("Traceability", "0", "0", "0"));
         Mockito.when(report.getToReviewSecurityHotspots()).thenReturn(listIssues);
         Assert.assertEquals(expected, SecurityHotspotsAdapter.getSecurityHotspotsByCategoryAndPriority(report));
     }

